src/agent/src/rpc.rs

```diff
@@ -1737,6 +1774,137 @@ fn load_kernel_module(module: &protocols::agent::KernelModule) -> Result<()> {
+        }
+
+        // not 100% sure we actually need this. the arg gets ignored anyway
+        pull_command
+            .arg("--decryption-key")
+            .arg("provider:attestation-agent:offline_fs_kbc::null")
+            .env("OCICRYPT_KEYPROVIDER_CONFIG", OCICRYPT_CONFIG_PATH);
+
+        //TODO: check if image is really encrypted since skopeo doesn't
+        //      seem to care
+    }
+
+    let status: ExitStatus = pull_command.status()?;
+
+    if !status.success() {
+        return Err(anyhow!(format!("failed to pull image: {:?}", status)));
+    }
+
+    // Copy image from one local file-system to another
+    // Resulting image is still stored in manifest format, but no longer includes the signature
+    // The image with a signature can then be unpacked into a bundle
+    let status: ExitStatus = Command::new(SKOPEO_PATH)
+        .arg("--insecure-policy")
+        .arg("copy")
+        .arg(&target_path_manifest)
+        .arg(&target_path_oci)
+        .arg("--remove-signatures")
+        .status()?;
+
+    if !status.success() {
+        return Err(anyhow!(format!("failed to copy image: {:?}", status)));
+    }
+
+    // To save space delete the manifest.
+    // TODO LATER - when verify image is added, this will need moving the end of that, if required
+    fs::remove_dir_all(&manifest_path)?;
+    Ok(())
+}
+
+fn unpack_image(cid: &str) -> Result<()> {
+    let source_path_oci = format!("/tmp/{}/image_oci:latest", cid);
+    let target_path_bundle = format!("{}{}{}", CONTAINER_BASE, "/", cid);
+
+    info!(sl!(), "cid is {:?}", cid);
+    info!(sl!(), "target_path_bundle is {:?}", target_path_bundle);
+
+    // Unpack image
+
+    let status: ExitStatus = Command::new(UMOCI_PATH)
+        .arg("--verbose")
+        .arg("unpack")
+        .arg("--image")
+        .arg(&source_path_oci)
+        .arg(&target_path_bundle)
+        .status()?;
+
+    if !status.success() {
+        return Err(anyhow!(format!("failed to unpack image: {:?}", status)));
+    }
+
+    // To save space delete the oci image after unpack
+    fs::remove_dir_all(&source_path_oci)?;
+
+    Ok(())
+}
+
+// TODO: what should we do if we can't spin up the AA?
+fn init_attestation_agent() {
+    // first write out the config
+    let config_path = OCICRYPT_CONFIG_PATH;
+    // Note that the image will need to be encrypted using a keyprovider
+    // that has the same name (at least according to the config).
+    let ocicrypt_config = serde_json::json!({
+        "key-providers": {
+            "attestation-agent":{
+                "grpc":format!("127.0.0.1:{}",AA_PORT)
+            }
+        }
+    });
+
+    let mut config_file = File::create(config_path).unwrap();
+    config_file
+        .write_all(ocicrypt_config.to_string().as_bytes())
+        .unwrap();
+
+    // start AA
+    // should we use tokio for this?
+    Command::new(AA_PATH)
+        .arg("--grpc_sock")
+        .arg(format!("127.0.0.1:{}", AA_PORT))
```

Comment:
Can you create a `const for `127.0.0.1` which is also hard-coded multiple times?